symengine/complex_mpc.cpp

```diff
@@ -908,6 +908,24 @@ class EvaluateMPC : public Evaluate
                 MPFR_RNDN);
         return complex_mpc(std::move(t));
     }
+    virtual RCP<const Basic> floor(const Basic &x) const override
+    {
+        SYMENGINE_ASSERT(is_a<ComplexMPC>(x))
+        mpc_class t(down_cast<const ComplexMPC &>(x).as_mpc().get_prec());
+        mpc_div_fr(t.get_mpc_t(),
+                   down_cast<const ComplexMPC &>(x).as_mpc().get_mpc_t(),
+                   MPFR_RNDDD);
```

Comment:
As far as I'm able to understand from the docs, `mpc_div_fr` rounds up in the direction of `MPFR_RNDxy`, where `x` and `y` are the rounding criteria for real and imaginary parts. `MPFR_RNDN` would simply round to the nearest integer, but we are concerned with `floor` and `ceiling`.